enumerable.rb

```diff
@@ -0,0 +1,157 @@
+# module for some enumerable methods
+module Enumerable
+  def my_each
+    if block_given?
+      for i in self
+        yield(i)
+      end
+      self
+    else
+      to_enum(:my_each)
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      for i in (0...length)
+        yield(self[i], i)
+      end
+      self
+    else
+      to_enum(:my_each_with_index)
+    end
+  end
+
+  def my_select
+    new_array = []
+    if block_given?
+      my_each { |num| new_array << num if yield(num) }
+      new_array
+    else
+      to_enum(:my_select)
+    end
+  end
+
+  def my_all?(pat = nil)
+    result = true
+    if block_given?
+      my_each { |x| result = false unless yield(x) }
+    elsif pat
+      my_each { |x| result = false unless pattern?(x, pat) }
+    else
+      my_each { |x| result = false unless x }
+    end
+    result
+  end
+
+  def my_any?(pat = nil)
+    result = false
+    if block_given?
+      my_each { |x| result = true if yield(x) }
+    elsif pat
+      my_each { |x| result = true if pattern?(x, pat) }
+    else
+      my_each { |x| result = true if x }
+    end
+    result
+  end
+
+  def my_none?(pat = nil)
+    if block_given?
+      !my_any? { |x| yield x }
+    elsif pat
+      !my_any?(pat)
+    else
+      my_each { |x| return false if x }
+      true
+    end
+  end
+
+  def my_count(src = nil)
+    contador = 0
+    if block_given?
+      my_each { |x| contador += 1 if yield x }
+    elsif src
+      my_each { |x| contador += 1 if x == src }
+    else
+      my_each { contador += 1 }
+    end
+    contador
+  end
+
+  def my_inject(*args)
+    x = args[0].nil? || args[0].class == Symbol ? 0 : args[0]
+    sym = args[0].class == Symbol ? args[0] : args[1].class == Symbol ? args[1] : nil
+    my_each do |elem|
+      x = if sym
+            x.send(sym, elem)
+          else
+            yield(x, elem)
+          end
+    end
+    x
```

Comment:
**[REQUIRED]**
* combines all elements of enum by applying a binary operation, specified by a block.
```
Failure/Error: let(:search) { proc { |memo, word| memo.length > word.length ? memo : word } }
     
     NoMethodError:
       undefined method `length' for 0:Integer
```